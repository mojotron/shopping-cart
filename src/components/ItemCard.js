import { Link } from "react-router-dom";
import "../styles/ItemCard.css";

const ItemCard = (props) => {
  return (
    <Link
      key={props.data.id}
      to={`/shop/${props.data.id}`}
      className="ItemCard__info__link"
    >
      <div className="ItemCard">
        <img
          className="ItemCard__image"
          src={props.data.images.icon}
          alt={props.data.name}
        />
        <div className="ItemCard__info">
          <h3 className="ItemCard__info__name">{props.data.name}</h3>
          <p className="ItemCard__info__type">{props.data.type.value}</p>
        </div>
      </div>
    </Link>
  );
};

export default ItemCard;
