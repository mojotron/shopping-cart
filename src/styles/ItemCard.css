.ItemCard {
  border-radius: 0.5rem;
  height: 30rem;
  width: 20rem;
  display: flex;
  flex-direction: column;
  background: linear-gradient(35deg, #141e30, #31557e);
  margin: 3rem;
  color: white;
  cursor: pointer;
}
.ItemCard:hover {
  box-shadow: rgb(18, 18, 19) 0px 20px 30px -10px;
}
.ItemCard__image {
  width: 130%;
  position: relative;
  bottom: 5rem;
  pointer-events: none;
}
.ItemCard__info {
  position: relative;
  bottom: 10rem;
  padding: 1rem;
  z-index: 10;
}
.ItemCard__info__name {
  font-size: 2rem;
}
.ItemCard__info__type {
  font-size: 1.7rem;
  margin-top: 1rem;
}
.ItemCard__info__link {
  display: block;
}
.layout-wrapper {
  padding: 10rem 5rem;
}
