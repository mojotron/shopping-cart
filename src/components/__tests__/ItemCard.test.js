import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import ItemCard from "../ItemCard";
import propsMock from "../../mocks/itemCardPropsData.json";

const MockItemCard = () => {
  return (
    <BrowserRouter>
      <ItemCard data={propsMock} />
    </BrowserRouter>
  );
};

describe("ItemCard component", () => {
  test("render ItemCard snapshot", () => {
    const { container } = render(<MockItemCard />);
    expect(container).toMatchSnapshot();
  });

  test("component is link", () => {
    render(<MockItemCard />);
    expect(screen.getByRole("link")).toHaveClass("ItemCard__info__link");
  });

  test("renders image", () => {
    render(<MockItemCard />);
    expect(screen.getByRole("img")).toBeInTheDocument();
  });

  test("render header with name of item", () => {
    render(<MockItemCard />);
    expect(screen.getByRole("heading").textContent).toEqual("temp");
  });

  test("paragraph renders item type", () => {
    render(<MockItemCard />);
    expect(screen.getByText("backpack")).toBeInTheDocument();
  });
});
