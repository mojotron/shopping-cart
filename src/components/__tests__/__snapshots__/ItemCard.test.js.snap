// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCard component render ItemCard snapshot 1`] = `
<div>
  <a
    class="ItemCard__info__link"
    href="/shop/1"
  >
    <div
      class="ItemCard"
    >
      <img
        alt="temp"
        class="ItemCard__image"
        src="/temp-path.jpg"
      />
      <div
        class="ItemCard__info"
      >
        <h3
          class="ItemCard__info__name"
        >
          temp
        </h3>
        <p
          class="ItemCard__info__type"
        >
          backpack
        </p>
      </div>
    </div>
  </a>
</div>
`;
